scripts/masterCron.ts

```diff
@@ -58,6 +64,115 @@ const getTopOfPeriod = async (provider: any, period: number) => {
   return topOfPeriod;
 };
 
+const decimalShift = async (collateralAsset: Contract) => {
+  let decimals = await collateralAsset.decimals();
+  return BigNumber.from(10).pow(BigNumber.from(18).sub(decimals));
+};
+
+async function getStrikePrice(
+  vault: Contract,
+  vaultLifecycle: Contract,
+  strikeSelection: Contract,
+  ierc20ABI: any
+) {
+  let expiry;
+  let currentOption = (await vault.optionState()).currentOption;
+  if (currentOption == address(0)) {
+    expiry = await vaultLifecycle.getNextFriday(
+      (
+        await provider.getBlock("latest")
+      ).timestamp
+    );
+  } else {
+    expiry = await vaultLifecycle.getNextFriday(
+      await new ethers.Contract(
+        currentOption,
+        ierc20ABI,
+        provider
+      ).expiryTimestamp()
+    );
+  }
+
+  let isPut = (await vault.vaultParams()).isPut;
+
+  let strike = await strikeSelection.getStrikePrice(expiry, isPut);
+
+  return expiry, isPut, strike;
+}
+
+async function getOptionPremium(
+  vault: Contract,
+  vaultLifecycle: Contract,
+  gnosisAuction: Contract,
+  optionsPremiumPricer: Contract,
+  strikePrice: BigNumber,
+  expiry: BigNumber,
+  isPut: boolean
+) {
+  let currentOption = (await vault.optionState()).currentOption;
+  let delay = await vault.delay();
+
+  let closeParams = {
+    OTOKEN_FACTORY: OTOKEN_FACTORY,
+    USDC: USDC_ADDRESS,
+    currentOption: currentOption,
+    delay: delay,
+    lastStrikeOverride: 0,
+    overriddenStrikePrice: 0,
+  };
+
+  let otokenAddress = await vaultLifecycle.getOrDeployOtoken(
+    closeParams,
+    await vault.vaultState(),
+    (
+      await vault.vaultParams()
+    ).underlying,
+    (
+      await vault.vaultParams()
+    ).asset,
+    strikePrice,
+    expiry,
+    isPut
+  );
+
+  let premium = await gnosisAuction.getOTokenPremium(
+    otokenAddress,
+    optionsPremiumPricer.address,
+    await vault.premiumDiscount()
+  );
+
+  return premium;
+}
+
+async function getAnnualizedVol(underlying: string, resolution: string) {
+  const latestTimestamp = (await provider.getBlock("latest")).timestamp;
+
+  var msg = {
+    jsonrpc: "2.0",
+    id: 833,
+    method: "public/get_volatility_index_data",
+    params: {
+      currency: underlying,
+      // resolution is in minutes, we multiply by 60 to get seconds
+      start_timestamp: (latestTimestamp - resolution * 60).toString(),
+      end_timestamp: latestTimestamp.toString(),
+      resolution: resolution,
+    },
+  };
+  var ws = new WebSocket("wss://www.deribit.com/ws/api/v2");
```

Comment:
Are you missing importing WebSocket from socket.io? Also shouldn't socket.io be tracked in package.json?

Reply:
don't think we need imports